include/eve/detail/function/simd/x86/swizzle.hpp

```diff
@@ -190,7 +190,7 @@ namespace eve::detail
           if constexpr(match(c,category::int64x4 ,category::uint64x4) )
           {
             auto vc = bit_cast(v, as_<as_floating_point_t<that_t>>{});
-            return bit_cast( v[q] , as(s));
+            return bit_cast( basic_swizzle(vc, q) , as(s));
```

Comment:
if you do this, then the swizzle won't be optimized by the find_optimized_pattern.
Is there a reason you changed that ?

Reply:
that was the point. 

We don't call main swizzle from the parts. Specifically that caused me a recursion when I fall back on basic swizzle.

Reply:
but then if this casted to float can be optimized, it wont ??

Reply:
If we are here, it's already couldn't be optimized.